src/scheduled_tasks.rs

```diff
@@ -450,6 +460,52 @@ async fn update_banned_when_expired(pool: &mut DbPool<'_>) {
   }
 }
 
+/// Find all unpublished posts with scheduled date in the future, and publish them.
+async fn publish_scheduled_posts(context: &Data<LemmyContext>) {
+  let pool = &mut context.pool();
+  let conn = get_conn(pool).await;
+
+  match conn {
+    Ok(mut conn) => {
+      let x: Option<Vec<(Post, Community)>> = post::table
+        .inner_join(community::table)
+        .filter(post::scheduled_time.is_not_null())
+        // TODO: throws error `overflow evaluating the requirement std::marker::Sized``
+        //.filter(post::scheduled_time.ge(Some(now())))
+        .get_results::<(Post, Community)>(&mut conn)
+        .await
+        .inspect_err(|e| error!("Failed to read unpublished posts: {e}"))
+        .ok();
+      let posts: Vec<(Post, Community)> = x
+        .into_iter()
+        .flatten()
+        .filter(|(p, _)| p.scheduled_time > Some(Utc::now()))
+        .collect();
+      let post_ids: Vec<PostId> = posts.iter().map(|p| p.0.id).collect();
```

Comment:
This whole block isn't necessary, collecting up the ids, to then run an update on them. 



You can just move your filters into the diesel update block below.

Reply:
Using a single query doesnt work because diesel [doesnt support joins in update queries](https://github.com/diesel-rs/diesel/issues/1478).

I rewrote the code here now.

Reply:
Mmmk you have to send the federation and webmention anyway, so a loop is required.